src/main/java/rx/schedulers/Schedulers.java

```diff
@@ -201,22 +197,44 @@ public static void reset() {
      */
     public static void shutdown() {
         Schedulers s = getInstance();
-        synchronized (s) {
-            if (s.computationScheduler instanceof SchedulerLifecycle) {
-                ((SchedulerLifecycle) s.computationScheduler).shutdown();
-            }
-            if (s.ioScheduler instanceof SchedulerLifecycle) {
-                ((SchedulerLifecycle) s.ioScheduler).shutdown();
-            }
-            if (s.newThreadScheduler instanceof SchedulerLifecycle) {
-                ((SchedulerLifecycle) s.newThreadScheduler).shutdown();
-            }
+        s.shutdownInstance();
 
+        synchronized (s) {
             GenericScheduledExecutorService.INSTANCE.shutdown();
-
+            
             RxRingBuffer.SPSC_POOL.shutdown();
 
             RxRingBuffer.SPMC_POOL.shutdown();
         }
     }
+    
+    /**
+     * Start the instance-specific schedulers.
+     */
+    synchronized void startInstance() {
```

Comment:
We try to avoid too much `private` modifiers; generally it adds hidden bridge methods, increasing the method count which is bad for Android. Also package level access leaves it open to same-package unit testing.